src/topology/category/Compactum.lean

```diff
@@ -0,0 +1,467 @@
+      { ext1,
+        exact is_ultrafilter.unique F.2 (ultrafilter.map X.str GG).2.1 this },
+      intros S hS,
+      refine h1 (or.inr ⟨S, by simpa⟩) },
+    rw [join_distrib, this],
+    exact ⟨h1 (or.inl rfl), rfl⟩ },
+  -- C2 is closed under finite intersections (by construction!).
+  have claim4 := finite_inter_closure_has_finite_inter C1,
+  -- C0 is closed under finite intersections by claim1.
+  have claim5 : has_finite_inter C0 := ⟨⟨set.univ, filter.univ_sets _, by simp⟩, claim1⟩,
+  -- Every element of C2 is nonempty.
+  have claim6 : ∀ P ∈ C2, (P : set (ultrafilter X)).nonempty,
+  { suffices : ∀ P ∈ C2, P ∈ C0 ∨ ∃ Q ∈ C0, P = AA ∩ Q,
+    { intros P hP,
+      cases this P hP,
+      { exact claim2 _ h },
+      { rcases h with ⟨Q, hQ, rfl⟩,
+        exact claim3 _ hQ } },
+    intros P hP,
+    exact claim5.finite_inter_closure_insert _ hP },
+  intros T hT,
+  -- Suffices to show that the intersection of the T's is contained in C2.
+  suffices : ⋂₀ ι T ∈ C2, by exact claim6 _ this,
+  -- Finish
+  apply claim4.finite_inter_mem,
+  intros t ht,
+  exact finite_inter_closure.basic (@hT t ht),
+end
+
+lemma is_closed_cl {X : Compactum} (A : set X) : is_closed (cl A) :=
+begin
+  rw is_closed_iff,
+  intros F hF,
+  exact cl_cl _ ⟨F, hF, rfl⟩,
+end
+
+lemma str_eq_of_le_nhds {X : Compactum} (F : ultrafilter X) (x : X) :
+  F.1 ≤ nhds x → X.str F = x :=
+begin
+  -- Notation to be used in this proof.
+  let fsu := finset (set (ultrafilter X)),
+  let ssu := set (set (ultrafilter X)),
+  let ι : fsu → ssu := coe,
+  let T0 : ssu := { S | ∃ A ∈ F.1, S = basic A },
+  let AA := (X.str ⁻¹' {x}),
+  let T1 := insert AA T0,
+  let T2 := finite_inter_closure T1,
+  intro cond,
+  -- If F contains a closed set A, then x is contained in A.
+  have claim1 : ∀ (A : set X), is_closed A → A ∈ F.1 → x ∈ A,
+  { intros A hA h,
+    by_contradiction,
+    rw le_nhds_iff at cond,
+    specialize cond Aᶜ a hA,
+    rw ultrafilter_iff_compl_mem_iff_not_mem.mp F.2 at cond,
+    contradiction },
+  -- If A ∈ F, then x ∈ cl A.
+  have claim2 : ∀ (A : set X), A ∈ F.1 → x ∈ cl A,
+  { intros A hA,
+    exact claim1 (cl A) (is_closed_cl A) (F.val.sets_of_superset hA (subset_cl A)) },
+  -- T0 is closed under intersections.
+  have claim3 : ∀ (S1 S2 ∈ T0), S1 ∩ S2 ∈ T0,
+  { rintros S1 S2 ⟨S1, hS1, rfl⟩ ⟨S2, hS2, rfl⟩,
+    exact ⟨S1 ∩ S2, F.1.inter_sets hS1 hS2, by simp [basic_inter]⟩ },
+  -- For every S ∈ T0, the intersection AA ∩ S is nonempty.
+  have claim4 : ∀ (S ∈ T0), (AA ∩ S).nonempty,
+  { rintros S ⟨S, hS, rfl⟩,
+    rcases claim2 _ hS with ⟨G, hG, hG2⟩,
+    exact ⟨G, hG2, hG⟩ },
+  -- Every element of T0 is nonempty.
+  have claim5 : ∀ (S ∈ T0), set.nonempty S,
+  { rintros S ⟨S, hS, rfl⟩,
+    exact ⟨F, hS⟩ },
+  -- Every element of T2 is nonempty.
+  have claim6 : ∀ (S ∈ T2), set.nonempty S,
+  { suffices : ∀ S ∈ T2, S ∈ T0 ∨ ∃ Q ∈ T0, S = AA ∩ Q,
+    { intros S hS,
+      specialize this _ hS,
+      cases this,
```

Comment:
```suggestion
      cases this _ hS,
```

Reply:
I would prefer to keep it as is, since `this` is used in the two branches generated by this `cases`.

Reply:
On second thought, I used `cases this _ hS with h h`, etc.